homework-7/whatsapp/views.py

```diff
@@ -0,0 +1,95 @@
+from whatsapp.app import app
+from whatsapp import utils
+from flask import jsonify, request, make_response
+import os
+import datetime
+
+
+@app.route('/')
+def home():
+    return "Home view"
+
+
+@app.route('/start-chat', methods=['POST'])
+def start_chat():
+    request_body = request.get_json()
+    # check inputs: format should be correct and user should not already exist
+    if 'username1' and 'username2' not in request_body:
+        return make_response(jsonify(error="The body must contain 'username1' and 'username 2' for sign up."), 400)
+    username1 = request_body['username1']
+    username2 = request_body['username2']
+
+    if utils.chat_exists(username1, username2):
+        return make_response(jsonify(error=f"Conversation between {username1} and {username2} already exists."), 400)
+
+    # create a directory and text file for the chat
+    chats_folder = f'whatsapp/chats/{username1}_{username2}'
+    try:
+        os.makedirs(chats_folder)
+        with open(os.path.join(chats_folder, f'{username1}_{username2}_chat.txt'), 'w'):
+            pass
+        return make_response(jsonify(message='ok'), 200)
+    except Exception as e:
+        print(f'error: {str(e)}')
+        return make_response(jsonify(error=str(e)), 500)
+
+
+@app.route('/send-message', methods=['POST'])
+def send_message():
+    request_body = request.get_json()
+    username1 = request_body['username1']
+    username2 = request_body['username2']
+    message = request_body['message']
+    sender = request_body['sender']
+    # inputs: username1, username2, message
+    # check if user is present
+    chats_folder = f'whatsapp/chats/{username1}_{username2}'
```

Comment:
If you decide to change the chats folder in the future, then you need to find all the places in the code where you're using it and modify it. The program can easily fail and create intricate bugs if you forget to modify this at some place.

A better approach is to define constants at one place and use them everywhere, eg: https://github.com/nitinl/redi-backend-spring-21/blob/main/flask-demo/whats_app/messaging.py#L5